public/app/features/alerting/unified/hooks/useProduceNewRuleGroup.test.tsx

```diff
@@ -0,0 +1,237 @@
+    setUpdateRulerRuleNamespaceHandler({ delay: 1000 });
+
+    render(<PauseTestComponent />);
+    expect(byText(/uninitialized/i).get()).toBeInTheDocument();
+
+    await userEvent.click(byRole('button').get());
+    expect(await byText(/loading/i).find()).toBeInTheDocument();
+
+    expect(await byText(/success/i).find()).toBeInTheDocument();
+    expect(await byText(/result/i).find()).toBeInTheDocument();
+    expect(byText(/error/i).query()).not.toBeInTheDocument();
+
+    const requests = await capture;
+    const [get, update, ...rest] = await serializeRequests(requests);
+
+    expect(update.body).toHaveProperty('rules[0].grafana_alert.is_paused', true);
+    expect([get, update, ...rest]).toMatchSnapshot();
+  });
+
+  it('should throw if the rule is not found in the group', async () => {
+    setUpdateRulerRuleNamespaceHandler();
+    render(
+      <PauseTestComponent
+        rulerRule={mockGrafanaRulerRule({ uid: 'does-not-exist', namespace_uid: grafanaRulerNamespace.uid })}
+      />
+    );
+    expect(byText(/uninitialized/i).get()).toBeInTheDocument();
+
+    await userEvent.click(byRole('button').get());
+    expect(await byText(/error: No rule with UID/i).find()).toBeInTheDocument();
+  });
+
+  it('should be able to handle error', async () => {
+    setUpdateRulerRuleNamespaceHandler({
+      delay: 1000,
+      response: new HttpResponse('oops', { status: 500 }),
+    });
+
+    render(<PauseTestComponent />);
+
+    expect(await byText(/uninitialized/i).find()).toBeInTheDocument();
+
+    await userEvent.click(byRole('button').get());
+    expect(await byText(/loading/i).find()).toBeInTheDocument();
+    expect(byText(/success/i).query()).not.toBeInTheDocument();
+    expect(await byText(/error: oops/i).find()).toBeInTheDocument();
+  });
+});
+
+describe('delete rule', () => {
+  it('should be able to delete a Grafana managed rule', async () => {
+    const rules = [
+      mockCombinedRule({
+        name: 'r1',
+        rulerRule: mockGrafanaRulerRule({ uid: 'r1' }),
+      }),
+      mockCombinedRule({
+        name: 'r2',
+        rulerRule: mockGrafanaRulerRule({ uid: 'r2' }),
+      }),
+    ];
+    const group = mockRulerRuleGroup({
+      name: 'group-1',
+      rules: [rules[0].rulerRule!, rules[1].rulerRule!],
+    });
+
+    const getGroup = rulerRuleGroupHandler({
+      delay: 1000,
+      response: HttpResponse.json(group),
+    });
+
+    const updateNamespace = updateRulerRuleNamespaceHandler({
+      response: new HttpResponse(undefined, { status: 200 }),
+    });
+
+    server.use(getGroup, updateNamespace);
+
+    const capture = captureRequests();
+
+    render(<DeleteTestComponent rule={rules[1]} />);
+
+    await userEvent.click(byRole('button').get());
+
+    expect(await byText(/success/i).find()).toBeInTheDocument();
+
+    const requests = await capture;
+    const serializedRequests = await serializeRequests(requests);
+    expect(serializedRequests).toMatchSnapshot();
```

Comment:
Nit: I did think for the snapshots that we could capture 2, to make it clear what the intent of each is:

```ts
const requestsOrder = serializedRequests.map((r) => [r.method, r.url.replace('http://localhost', '')]);
const requestBodies = serializedRequests.map((r) => r.body);
expect(requestsOrder).toMatchSnapshot('request endpoints are called in correct order');
expect(requestBodies).toMatchSnapshot('request bodies are correct');
```

Then you get one that looks like this, which makes it clear to see exactly what we're expect (without the body/headers potentially muddying the waters)

```
exports[`delete rule should be able to delete a Grafana managed rule: request endpoints are called in correct order 1`] = `
[
  [
    "GET",
    "/api/ruler/grafana/api/v1/rules/NAMESPACE_UID/mockCombinedRuleGroup?subtype=cortex",
  ],
  [
    "POST",
    "/api/ruler/grafana/api/v1/rules/NAMESPACE_UID?subtype=cortex",
  ],
  [
    "GET",
    "/api/ruler/grafana/api/v1/rules/NAMESPACE_UID/mockCombinedRuleGroup?subtype=cortex",
  ],
]
`;
```